oioioi/programs/handlers.py

```diff
@@ -555,6 +555,9 @@ def make_report(env, kind='NORMAL', save_scores=True, **kwargs):
         test_report.score = result['score'] if save_scores else None
         test_report.status = result['status']
         test_report.time_used = result['time_used']
+        percentage = result.get('result_percentage', 0.)
+        if percentage != 100. and percentage != 0.:
```

Comment:
Can You please convince me it is okay here to check equality of some floating point number? I would prefer to rather see
```
if percentage > 0 and percentage < 100:
```
keep in mind that sometimes "different" number are equal due to precision
```
>>> 100.000000000000001 == 100.
True
>>> 100.00000000000001 == 100.
False
```

Reply:
Also, the domain of these values is not necessarily `[0, 100]`. I actually have abused this mechanism with values well above 100 to circumvent some architectural shortcomings once.

Reply:
Thanks for the insight. In the wiki it says that the percentage is a whole integer. It says that this option can be used to give partial scoring. It does not mention whether it is allowed to grant more than 100%.  It should be more documented in the specs

As for this logic, I am not sure we should tell user that he got 150% for this test, as this i probably not the intended behavior for this feature and was properly called "an abuse".

I suspect there would be disagreements about this, so please express your opinions here!

Reply:
I don't know if the percentage value should be greater than 100, but in the context of this PR I think that the message should not be displayed when the value is 0 (then the result should be WA) or 100 (then there is no comment and you can see assume that you have full points).

But if we decided to allow a percentage greater than 100, we would have to find another way to check whether this value is equal to 100

Reply:
First of all, if we could change this to an Integer, the float-related discussion is unnecessary. Is there any reason to use `FloatField` not `IntegerField`?

Secondly I think that Verdict and partial scoring are different things and it should be allowed to give 0% points but OK as a verdict. The checker returns both the verdict and partial scoring independently. We should treat is as such independent values.

Thirdly If partial scoring does not affect number of points awarded there should be no comment (what for it would be). If user get less points we definitely should post the comment. If it is allowed to give more, it is debatable if it should be displayed.